nevergrad/benchmark/experiments.py

```diff
@@ -573,6 +595,17 @@ def yabbob(seed: tp.Optional[int] = None, parallel: bool = False, big: bool = Fa
         for num_blocks in [1]
         for d in ([100, 1000, 3000] if hd else [2, 10, 50])
     ]
+    max_num_constraints = 4
+    constraints_list: tp.List[tp.Any] = [_Constraint(name, as_bool)
+        for as_bool in [False, True]
+        for name in ["positive_sum", "positive_diff", "positive_second_diff", "ball"]
+        ]
+    assert constraints < len(constraints_list) + max_num_constraints, (
+        "constraints should be in 0, 1, ..., {len(constraints_list) + max_num_constraints - 1} (0 = no constraint).")
+    for func in functions:
+        for u in range(len(constraints_list)):
+            if u < constraints and constraints - max_num_constraints <= u:  # we add the constraints with index u in {constraints-max, ..., constraints-1}
+                func.parametrization.register_cheap_constraint(constraints_list[u % len(constraints_list)])
```

Comment:
this is definitely still very confusing for me
`constraint_list` should probably just be `constraints`
then `constraints` should be something else but I can't get a hold of what it would mean `constraint_case` maybe?

Also do you need to add a descriptor for the cases? not sure

Reply:
There is already a variable termed constraints.

Reply:
I try to refactor a little bit.

Reply:
yes exactly, my point is "why". I don't understand *what* `constraints` currently represent, so the name should maybe be changed to clarify

Reply:
Yo! Looks better.